src/css-parser.js

```diff
@@ -0,0 +1,193 @@
+
+*/
+CSSParser.cssReplaceChar = [ '-', '\\', '_', ':', ';', '(', ')', '"', '\'', '\/', ',', '+', '%', '#', '!', '*', ' ' ];
+CSSParser.reCss = [
+    /&#0*8722;?|&#[xX]0*2212;?|&minus;|&#0*8208;?|&#[xX]0*2010;?|&dash;|&hyphen;/g,
+    /&#0*92;?|&#[xX]0*5[cC];?|&bsol;/g,
+    /&#0*95;?|&#[xX]0*5[fF];?|&lowbar;|&UnderBar;/g,
+    /&#0*58;?|&#[xX]0*3[aA];?|&colon;/g,
+    /&#0*59;?|&#[xX]0*3[bB];?|&semi;/g,
+    /&#0*40;?|&#[xX]0*28;?|&lpar;/g,
+    /&#0*41;?|&#[xX]0*29;?|&rpar;/g,
+    /&#0*34;?|&#[xX]0*22;?|&quot;|&QUOT;/g,
+    /&#0*39;?|&#[xX]0*27;?|&apos;/g,
+    /&#0*47;?|&#[xX]0*2[fF];?|&sol;/g,
+    /&#0*44;?|&#[xX]0*2[cC];?|&comma;/g,
+    /&#0*43;?|&#[xX]0*2[bB];?|&plus;/g,
+    /&#0*37;?|&#[xX]0*25;?|&percnt;/g,
+    /&#0*35;?|&#[xX]0*23;?|&num;/g,
+    /&#0*33;?|&#[xX]0*21;?|&excl;/g,
+    /&#0*42;?|&#[xX]0*2[aA];?|&ast;|&midast;/g,
+    /&#0*32;?|&#[xX]0*20;?|&#0*9;?|&#[xX]0*9;?|&Tab;|&#0*10;?|&#[xX]0*[aA];?|&NewLine;|&#0*12;?|&#[xX]0*[cC];?|&#0*13;?|&#[xX]0*[dD];?|\t|\r|\n|\f/g,
+];
+
+CSSParser.reExpr = /(['"]?)?\s*([0-9a-z%#\-+_,!\/\\\*]+)(\([^\(]?\)?)?\s*(['"]?)?\s*/ig;
+
+/* re for URL pattern */
+CSSParser.reUrlUnquoted = /^url\(\s*$/i;
+CSSParser.reUrlSingleQuoted = /^url\(\s*'$/i;
+CSSParser.reUrlDoubleQuoted = /^url\(\s*"$/i;
+
+/* emum type of parseStyleAttributeValue */
+CSSParser.STYLE_ATTRIBUTE_URL_UNQUOTED      = 1;
+CSSParser.STYLE_ATTRIBUTE_URL_SINGLE_QUOTED = 2;
+CSSParser.STYLE_ATTRIBUTE_URL_DOUBLE_QUOTED = 3;
+CSSParser.STYLE_ATTRIBUTE_UNQUOTED          = 4;
+CSSParser.STYLE_ATTRIBUTE_SINGLE_QUOTED     = 5;
+CSSParser.STYLE_ATTRIBUTE_DOUBLE_QUOTED     = 6;
+
+CSSParser.STYLE_ATTRIBUTE_ERROR_AT_PROP_LOCATION = -1;
+CSSParser.STYLE_ATTRIBUTE_ERROR_PROP_EMPTY       = -2;
+CSSParser.STYLE_ATTRIBUTE_ERROR                  = -3;
+
+/**
+* @function CSSParser.htmlStyleAttributeValueEntitiesDecode
+*
+* @description
+* htmlStyleAttributeValueEntitiesDecode for delimiter in style attribute as defined in the spec.
+* all special characters being defined must be decoded first (defined in cssReplaceChar).
+* http://www.w3.org/TR/css-style-attr/
+*/
+CSSParser.htmlStyleAttributeValueEntitiesDecode = function(str) {
+    /* html decode the str before CSS parsing,
+       we follow the parsing order of the browser */
+    var len = CSSParser.reCss.length;
+    for (var j=0;j<len;++j) {
+        str = str.replace(CSSParser.reCss[j], CSSParser.cssReplaceChar[j]);
+    }
+    return str;
+};
+
+/**
+* @function CSSParser.parseStyleAttributeValue
+*
+* @description
+*/
+CSSParser.parseStyleAttributeValue = function(str) {
+    var r = { prop: '', code: NaN };
+    var kv = str.split(';'); // it will return new array even there is no ';' in the string
+    var v = kv[kv.length-1].split(':'); // only handling the last element
+    if (v.length === 2) {
+
+        // it is ok to trim the space at both ends
+        var prop = v[0].trim(), 
+            expr = v[1].trim();
+        if (prop === '') {
```

Comment:
String.trim() is not supported by IE 8 or below
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/Trim